internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -678,7 +694,7 @@ func (s *Server) processRelevantRepositoryEvent(
 	}
 
 	// This only makes sense for "meta" event type
-	if dbrepo.WebhookID.Valid {
+	if event.GetHookID() != 0 && dbrepo.WebhookID.Valid {
 		// Check if the payload webhook ID matches the one we
 		// have stored in the DB for this repository
 		if event.GetHookID() != dbrepo.WebhookID.Int64 {
```

Comment:
Does the comment on line 701-703 still make sense?

Reply:
I think it does.

This routine is called only on `meta` and `repository` events, and this fix ensures that we check the hook id only when available, which is the case solely for the `meta` event.

The comment per sé is still valid, but can be removed if it's not useful.
I lean towards leaving it there for posterity, maybe rewriting it at the top of the `if`, but let me know.

Reply:
I agree that it's valid and useful